code/modules/research/xenobiology/crossbreeding/warping

```diff
@@ -0,0 +1,796 @@
+		cooldown = world.time + max_cooldown
+		teleport_fire() //triggers it every 5 seconds anyway
+
+
+
+
+///teleport people and put them on fire if they run into the rune.
+/obj/effect/warped_rune/orangespace/proc/teleport_fire()
+	if(!locate(/obj/effect/hotspot) in T) //doesn't teleport item but put them on fire anyway for good measure.
+		new /obj/effect/hotspot(T)
+
+	for(var/mob/living/F in T)
+		do_teleport(F, T, 3, asoundin = 'sound/effects/phasein.ogg', channel = TELEPORT_CHANNEL_BLUESPACE)//only set the teleport to one for now
+		F.adjust_fire_stacks(10)
+		F.IgniteMob()
+
+/*
+		for(F.fire_stacks > 0 && !istype(F,/mob/living/simple_animal)) //Keeps teleporting the guy as long as he is on fire (doesn't include simplemobs) WIP
+			do_teleport(F, get_turf(F), 5, asoundin = 'sound/effects/phasein.ogg', channel = TELEPORT_CHANNEL_BLUESPACE)
+ */
+
+
+
+
+
+
+
+
+
+
+/*The purple warp rune makes suture and ointment if you put cloth or plastic on it. */
+
+
+
+
+/obj/item/slimecross/warping/purple
+	colour = "purple"
+	runepath = /obj/effect/warped_rune/purplespace
+	effect_desc = "creates a rune that transforms plastic into suture and cloth into regenerative mesh"
+
+/obj/effect/warped_rune/purplespace
+	desc = "When all that was left were plastic walls and the clothes on their back, they knew what they had to do."
+	icon_state = "purplespace"
+	var/obj/item/stack/medical/suture/S          // S stands for suture because it just make sense
+	var/obj/item/stack/medical/mesh/M            //M is for mesh and you
+	max_cooldown = 30
+
+/obj/effect/warped_rune/purplespace/Initialize()
+	.=..()
+	START_PROCESSING(SSprocessing, src)
+
+
+/obj/effect/warped_rune/purplespace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		transmute_heal()
+
+
+///""transforms"" cloth and plastic into suture and regenerative mesh
+/obj/effect/warped_rune/purplespace/proc/transmute_heal() //transmute plastic and cloth into suture and regenerative mesh
+	for(var/obj/item/stack/sheet/plastic/P in T)  //transmute Plastic into regenerative mesh
+		M = new (T,1)
+		P.amount--
+		playsound(T, 'sound/effects/splat.ogg', 20, TRUE)
+		if(P.amount <= 0)
+			qdel(P)
+
+	for(var/obj/item/stack/sheet/cloth/C in T) //transmute cloth into suture
+		S = new(T, 1)
+		C.amount--
+		playsound(T, 'sound/effects/splat.ogg', 20, TRUE)
+		if(C.amount <= 0)
+			qdel(C)
+
+
+
+
+
+
+/* the blue warp rune  keeps a tile slippery CONSTANTLY by adding water (maybe I'll make it lube if I feel extra fucky) every few(10) seconds */
+
+
+
+/obj/item/slimecross/warping/blue
+	colour = "blue"
+	runepath = /obj/effect/warped_rune/cyanspace //we'll call the blue rune cyanspace to not mix it up with actual bluespace rune
+	effect_desc = "creates a rune that constantly wet itself with water as long as the rune is up"
+
+/obj/effect/warped_rune/cyanspace
+	icon_state = "slipperyspace"
+	desc = "Water drowns, water feeds, water slips, is there a thing water cannot do?"
```

Comment:
might wanna turn that comma after slips into a period or a semicolon or something, idk (it might be fine as it is now)

Reply:
also, replace "a thing" with "anything"